### PR TITLE
Clear the docker setup - Deprecate post-create.sh and use bin/setup

### PR DESCRIPTION
**Pull request by fabiosammy**
Not sure why this is here. The original PR does not explain the benefits of having the assets ready (https://github.com/mastodon/mastodon/pull/22277), it mentions an error on running the tests, but you can build the assets before run the tests. The assets precompile adds an overhead, and make it a little confusing when you're contributing with the development (we have the webpack dev server running).

Also, I added the `gem install bundler` based on the version from the Gemfile.lock instead of reverting the changes on the Gemfile.lock

### CODE REVIEW

File: .devcontainer/docker-compose.yml [hunk 1]
```diff
@@ -1,5 +1,3 @@
-version: '3'
```
Comment (fabiosammy) [inline .devcontainer/docker-compose.yml:1]: This is deprecated.

**Comment by mjankowski** [comment]
I realize this is a tangent to your changes here, but -- could we delete the `post-create.sh` script and rely on the existing `bin/setup` script instead? ... or are there perms/other issues with that?

Strikes me that conceptually the end state of "you are doing local development without Docker or a devcontainer and have configured all dependencies and services by yourself" and "you have used the devcontainer config to get everything in place in a Docker setup" should both be in roughly the same place in terms of wanting to run `bin/setup` as the next step before doing any actual development?

File: .devcontainer/post-create.sh [hunk 2]
```diff
@@ -2,26 +2,17 @@
...
 
-# [re]create, migrate, and seed the test database
-RAILS_ENV=test ./bin/rails db:setup
-
 # [re]create, migrate, and seed the development database
-RAILS_ENV=development ./bin/rails db:setup
-
-# Precompile assets for development
-RAILS_ENV=development ./bin/rails assets:precompile
+./bin/rails db:setup
```
Comment (mjankowski) [inline .devcontainer/post-create.sh:17]: We have this (atypical for rails apps) hard check for the presence of RAILS_ENV (See boot.rb), and I think this command will fail as-is?

My personal preference would be to remove that check, but apparently it has helped stop bad things from happening to good people.

Comment (fabiosammy) [reply .devcontainer/post-create.sh:17]: So, in the "docker-compose.yml" we already have the RAILS_ENV defined there.

**Comment by fabiosammy** [comment]
@mjankowski totally agree! So, I removed the entire post-create script and make the references to the bin/setup

**Review by renchap — APPROVED** [review]

**Comment by renchap** [comment]
Thanks for this, this is indeed a nice simplification!

**Pull request merged**